### PR TITLE
Utilize source maps for error improvements

### LINKED ISSUE #3966: Utilize source maps
As part of adding code frames for the [improved error experience](https://github.com/cypress-io/cypress/issues/3762), we need to do the following:

* Enable source maps (unless the user has explicitly disabled them) in the preprocessor
* Gracefully handle the case where the user has disabled source maps
* For errors with a stack:
  - Translate the stack into source code stack
* For command errors:
  - Capture the stack when a command runs
  - If the command fails, use the source maps to trace the command back to the source spec file
  - Pull out the line from the spec file and one two above and below it to display in the error code frame
* For assertion errors:
  - Same as commands but point to and show the assertion
* For reporter UI:
  - Capture the stack when an `it` runs
  - Display a button in the UI next to the test name that opens the file at that line when clicked

Related: #881

### PR DESCRIPTION
**Pull request by chrisbreiding**
Closes #3966

### CODE REVIEW

**Comment by jennifer-shehane** [comment]
Hmmm, wonder if we should converge on a PR to work on together for this.

**Comment by ChaseBig** [comment]
How are we coming along with this PR? I've spent close to 3hr's trying to debug a series of long-form file tests that each have similar actions because I don't know where in the code the test failed. The test is for a continuous "information form" feature that has 5-10 form questions on 10 different pages that each have to be done in succession, thus meaning I cannot split the test up into individual smaller tests. It is soooo frustrating having to spend so much time just finding the line of code that a test failed on.

**Comment by jennifer-shehane** [comment]
@ChaseBig There's work being done every day on this. It's a complicated problem to solve, so we don't have a clear estimate for delivery.

**Comment by jennifer-shehane** [comment]
```
[desktop-gui:build  ] ERROR in ../ui-components/src/select/select.scss
[desktop-gui:build  ] Module build failed (from ../web-config/node_modules/mini-css-extract-plugin/dist/loader.js):
[desktop-gui:build  ] ModuleBuildError: Module build failed (from ../web-config/node_modules/sass-loader/lib/loader.js):
[desktop-gui:build  ] 
[desktop-gui:build  ]     @extend .#{$fa-css-prefix}-circle-o;
[desktop-gui:build  ]              ^
[desktop-gui:build  ]       Undefined variable: "$fa-css-prefix".
[desktop-gui:build  ]       in /root/cypress/packages/ui-components/src/select/select.scss (line 13, column 15)
[desktop-gui:build  ]     at /root/cypress/packages/desktop-gui/node_modules/webpack/lib/NormalModule.js:302:20
[desktop-gui:build  ]     at /root/cypress/packages/desktop-gui/node_modules/loader-runner/lib/LoaderRunner.js:367:11
[desktop-gui:build  ]     at /root/cypress/packages/desktop-gui/node_modules/loader-runner/lib/LoaderRunner.js:233:18
[desktop-gui:build  ]     at context.callback (/root/cypress/packages/desktop-gui/node_modules/loader-runner/lib/LoaderRunner.js:111:13)
[desktop-gui:build  ]     at Object.callback (/root/cypress/packages/web-config/node_modules/sass-loader/lib/loader.js:52:13)
[desktop-gui:build  ]     at Object.done [as callback] (/root/cypress/packages/web-config/node_modules/neo-async/async.js:8067:18)
[desktop-gui:build  ]     at options.error (/root/cypress/packages/web-config/node_modules/node-sass/lib/index.js:294:32)
```

File: packages/desktop-gui/src/settings/settings.jsx [hunk 1]
```diff
@@ -34,6 +35,9 @@ const Settings = observer(({ project, app }) => {
           <Panel header='Proxy Settings' key='proxy-settings' className='form-horizontal settings-proxy'>
             <ProxySettings app={app} />
           </Panel>
```
Comment (flotwig) [inline packages/desktop-gui/src/settings/settings.jsx]: "Proxy Settings" is technically a "Global Setting", at some point, we should move the preferred editor + proxy settings to their own "Global Settings" section. 

So we don't overly inflate the UI changes from this PR, maybe the "Proxy Settings" should just be moved into the new "Global Settings" section for now?

**Pull request merged**